app_flutter/lib/build_dashboard.dart

```diff
@@ -11,29 +11,60 @@ import 'status_grid.dart';
 
 /// [BuildDashboard] parent widget that manages the state of the dashboard.
 class BuildDashboardPage extends StatefulWidget {
+  BuildDashboardPage({FlutterBuildState buildState})
+      : buildState = buildState ?? FlutterBuildState();
+
+  final FlutterBuildState buildState;
+
+  @visibleForTesting
+  static const Duration errorSnackbarDuration = Duration(seconds: 8);
   @override
   _BuildDashboardPageState createState() => _BuildDashboardPageState();
 }
 
 class _BuildDashboardPageState extends State<BuildDashboardPage> {
-  final FlutterBuildState buildState = FlutterBuildState();
+  final GlobalKey<ScaffoldState> _scaffoldKey = GlobalKey<ScaffoldState>();
+
+  FlutterBuildState buildState;
 
   @override
   void initState() {
     super.initState();
 
+    buildState = widget.buildState;
```

Comment:
Why do you capture the buildState here? If the widget rebuilds with a new BuildState object you will hold on to and old reference

Reply:
Thanks for pointing this out! I moved this into the build method since it's just updating a reference.